site/book/05-developing-functions/02-developing-in-Go.md

```diff
@@ -13,121 +12,161 @@ the following features:
 ## Quickstart
 
-### Create the go module
+### Initialize your project
+
+We start from a "get-started" package which contains a `main.go` file with some scaffolding code.
 
 ```shell
-$ go mod init github.com/user/repo; go get sigs.k8s.io/kustomize/kyaml@v0.12.0
-```
+export FUNCTION_NAME=<your KRM function name>
 
-### Create the `main.go`
+# Get the "get-started" package.
+kpt pkg get https://github.com/GoogleContainerTools/kpt-functions-sdk.git/go/get-started@master ${FUNCTION_NAME}
 
-This is a simple function that adds the annotation `myannotation` with the
-provided value:
+cd ${FUNCTION_NAME}
+
+# Initialize the Go module
+go mod init
+go mod tidy
+```
+
+### Write the KRM function logic
+ 
+Take a look at the `main.go` (as below) and complete the `Run` function.
 
 ```go
 // main.go
 package main
 
 import (
-	"os"
+  "fmt"
+  "os"
 
-	"sigs.k8s.io/kustomize/kyaml/fn/framework"
-	"sigs.k8s.io/kustomize/kyaml/fn/framework/command"
-	"sigs.k8s.io/kustomize/kyaml/kio"
-	"sigs.k8s.io/kustomize/kyaml/yaml"
+  "github.com/GoogleContainerTools/kpt-functions-sdk/go/fn"
 )
 
+// EDIT THIS FUNCTION!
+// This is the main logic. rl is the input `ResourceList` which has the `FunctionConfig` and `Items` fields.
+// You can modify the `Items` and add result information to `rl.Results`.
+func Run()(rl *fn.ResourceList) (bool, error) {
+// Your code
+}
+
 func main() {
-	// create a struct matching the structure of ResourceList.FunctionConfig to hold its data
-	var config struct {
-		Data map[string]string `yaml:"data"`
-	}
-	fn := func(items []*yaml.RNode) ([]*yaml.RNode, error) {
-		for i := range items {
-			// set the annotation on each resource item
-			err := items[i].PipeE(yaml.SetAnnotation("myannotation", config.Data["myannotation"]))
-			if err != nil {
-				return nil, err
-			}
-		}
-		return items, nil
-	}
-	p := framework.SimpleProcessor{Filter: kio.FilterFunc(fn), Config: &config}
-	cmd := command.Build(p, command.StandaloneDisabled, false)
-  // Adds a "gen" subcommand to create a Dockerfile for building the function into a container image.
-  command.AddGenerateDockerfile(cmd)
-
-	if err := cmd.Execute(); err != nil {
-		os.Exit(1)
-	}
+  // CUSTOMIZE IF NEEDED
+  // `AsMain` accepts a `ResourceListProcessor` interface.
+  // You can explore other `ResourceListProcessor` structs in the SDK or define your own.
+  if err := fn.AsMain(fn.ResourceListProcessorFunc(Run)); err != nil {
+    os.Exit(1)
+  }
 }
 ```
 
-### Build and test the function
+The `fn` library provides a series of KRM level operations for `ResourceList`. 
+Basically, the KRM resource `ResourceList.FunctionConfig` and KRM resources `ResourceList.Items` are both converted to 
+`KubeObject` objects. You can use `KubeObject` similar as [`unstructured.Unstrucutred`](https://pkg.go.dev/k8s.io/apimachinery/pkg/apis/meta/v1/unstructured).
 
-Build the go binary:
+To learn more about the `KubeObject`, see https://pkg.go.dev/github.com/GoogleContainerTools/kpt-functions-sdk/go/fn
```

Comment:
The examples (https://github.com/GoogleContainerTools/kpt-functions-sdk/tree/master/go/fn/examples) are not shown up in the go doc after we re-organise it to fix the k8s dependencies.
I'd suggest add a link to the examples since they cover several different use cases.

Reply:
The examples are in https://pkg.go.dev/github.com/GoogleContainerTools/kpt-functions-sdk/go/fn/examples 

Do you mean they should be shown under each function?